client/clients/router/client.go

```diff
@@ -367,22 +397,53 @@ func (c *Cli) updateConnection(ctx context.Context) {
 	cc, url := c.getLeaderClientConn()
 	if cc == nil || len(url) == 0 {
 		log.Warn("[router] got an invalid leader client connection", zap.String("url", url))
-		return
 	}
 	if c.conCtxMgr.Exist(url) {
 		log.Debug("[router] the router leader remains unchanged", zap.String("url", url))
-		return
-	}
-	stream, err := pdpb.NewPDClient(cc).QueryRegion(ctx)
-	if err != nil {
-		log.Error("[router] failed to create the router stream connection", errs.ZapError(err))
+	} else if cc != nil && len(url) > 0 {
+		stream, err := pdpb.NewPDClient(cc).QueryRegion(ctx)
+		if err != nil {
+			log.Error("[router] failed to create the leader router stream connection", errs.ZapError(err))
+		}
+		// Store the stream connection context if it is successfully created.
+		if stream != nil {
+			c.conCtxMgr.Store(ctx, url, stream)
+			log.Info("[router] successfully established the leader router stream connection", zap.String("url", url))
+		}
 	}
-	// Store the stream connection context if it is successfully created.
-	if stream != nil {
-		c.conCtxMgr.Store(ctx, url, stream)
+	// If enabled the follower handle, we need to update the follower router stream connections as well.
+	if c.option.GetEnableFollowerHandle() {
+		conns := c.getAllClientConns()
+		if len(conns) == 0 {
+			log.Warn("[router] no router node found")
+			return
+		}
+		for url, conn := range conns {
+			if c.conCtxMgr.Exist(url) {
+				log.Debug("[router] the router node remains unchanged", zap.String("url", url))
+				continue
+			}
+			stream, err := pdpb.NewPDClient(conn).QueryRegion(ctx)
+			if err != nil {
+				log.Error("[router] failed to create the router stream connection", errs.ZapError(err))
+			}
+			// Store the stream connection context if it is successfully created.
+			if stream != nil {
+				c.conCtxMgr.Store(ctx, url, stream)
+				log.Info("[router] successfully established the router stream connection", zap.String("url", url))
+			}
+		}
+	} else {
+		// GC all the follower router stream connections.
+		c.conCtxMgr.GC(func(url string) bool {
```

Comment:
Do we need the handle the scale in/out cases?

Reply:
Nice catch. Added.